index.js

```diff
@@ -179,6 +181,16 @@ function processDecompressParams (opts) {
 }
 
 function buildRouteCompress (fastify, params, routeOptions, decorateOnly) {
+  // This methods works by altering the routeOptions, it has side effects.
```

Comment:
I don't understand how this is getting added twice and why the side effect is necessary. The test is only registering the plugin once. Can you provide detail?

Reply:
It should be caused by the `ignoreTrailingSlash` that register the routes `/hello` and `/hello/`

Reply:
i Confirm : 
```
/test
/test/
```


```js
    function afterRouteAdded (path, notHandledErr, done) {
      const url = prefix + path
      console.log(url)
```

Reply:
Excellent. That reduces this to the actual issue.

Reply:
This might be good to be documented in https://www.fastify.io/docs/latest/Hooks/#onroute don't you think? Like a `Note`.